From 54a5fbe126d2a28913a81f7191d7ac6db70e1f43 Mon Sep 17 00:00:00 2001
From: Chet Ramey <chet.ramey@case.edu>
Date: Mon, 12 Oct 2015 13:57:17 +0000
Subject: bash-4.4 beta release

---
(limited to 'input.c')

diff --git a/input.c b/input.c
index 14c6f4e..308b87e 100644
--- a/input.c
+++ b/input.c
@@ -205,7 +205,7 @@ make_buffered_stream (fd, buffer, bufsize)
   if (bufsize == 1)
     bp->b_flag |= B_UNBUFF;
   if (O_TEXT && (fcntl (fd, F_GETFL) & O_TEXT) != 0)
-    bp->b_flag |= O_TEXT;
+    bp->b_flag |= B_TEXT;
   return (bp);
 }
 
@@ -272,7 +272,7 @@ save_bash_input (fd, new_fd)
       return -1;
     }
 
-  if (buffers[nfd])
+  if (nfd < nbuffers && buffers[nfd])
     {
       /* What's this?  A stray buffer without an associated open file
         descriptor?  Free up the buffer and report the error. */
--
cgit v0.9.0.2

